tests/test_aws.py

```diff
@@ -12,26 +14,34 @@ class TestAWSConnectionString(unittest.TestCase):
     LOCAL_CONNECTION_STRING = 'aws_access_key_id=aws_mock_key;aws_secret_access_key=aws_mock_secret'
     EC2_CONNECTION_STRING = 'aws_access_key_id=aws_mock_key;aws_secret_access_key=aws_mock_secret;token=aws_mock_token'
 
-    @patch.object(boto, 'connect_s3')
-    def test_from_local_authentication(self, mock_boto_connection_request):
-        mock_aws_connection = Mock()
-        mock_aws_connection.get_bucket.return_value = None
+    @patch.object(boto3, 'resource')
+    @patch.object(boto3, 'Session')
+    def test_from_local_authentication(self, mock_boto_session, mock_boto_connection_request):
+        mock_aws_credentials = Mock()
         key_property = PropertyMock(return_value='aws_mock_key')
         secret_property = PropertyMock(return_value='aws_mock_secret')
-        type(mock_aws_connection).aws_access_key_id = key_property
-        type(mock_aws_connection).aws_secret_access_key = secret_property
-        mock_boto_connection_request.return_value = mock_aws_connection
+        type(mock_aws_credentials).access_key = key_property
+        type(mock_aws_credentials).secret_key = secret_property
+        mock_boto_session.return_value.get_credentials.return_value = mock_aws_credentials
 
         connection_string = AWS().connection_string()
 
         self.assertEqual(connection_string, self.LOCAL_CONNECTION_STRING)
 
     @patch.object(AWS, '_request_temporary_credentials')
-    @patch.object(boto, 'connect_s3')
-    def test_from_ec2_iam_authentication(self, mock_boto_connection_request, mock_aws_request):
-        mock_aws_connection = Mock()
-        mock_aws_connection.get_bucket.side_effect = boto.exception.S3ResponseError(status=403, reason='forbidden')
-        mock_boto_connection_request.return_value = mock_aws_connection
+    @patch.object(boto3, 'Session')
+    @patch.object(boto3, 'resource')
+    def test_from_ec2_iam_authentication(self, mock_boto_resource, mock_boto_session, mock_aws_request):
```

Comment:
this test... :/  the alternative to such a brittle test is way more architecture around something we don't need. the tradeoffs are right, i'm just sad about it.